include/gtk-layer-shell.h

```diff
@@ -312,23 +338,23 @@ gboolean gtk_layer_auto_exclusive_zone_is_enabled (GtkWindow *window);
 /**
  * gtk_layer_set_keyboard_interactivity:
  * @window: A layer surface.
- * @interacitvity: Whether the layer surface should receive keyboard events.
+ * @interactivity: The type of keyboard interactivity requested.
  *
  * Whether the @window should receive keyboard events from the compositor.
  *
- * Default is %FALSE
+ * Default is #GTK_LAYER_SHELL_KEYBOARD_NONE
  */
-void gtk_layer_set_keyboard_interactivity (GtkWindow *window, gboolean interacitvity);
+void gtk_layer_set_keyboard_interactivity (GtkWindow *window, GtkLayerShellKeyboardInteractivity interactivity);
```

Comment:
This and the below function break backwards compatibility. We'll want to deprecate the old functions but keep them around.

Reply:
Since true == 1 and false == 0, this breaks ABI but doesn't break API. Depending on the project policy, this may be okay.

Reply:
I am trying to keep the ABI stable (various versions are shipped with various distros, and apps should ideally work with all of them). It *might* even be ABI-compatible (enum types seem to be implementation defined), but no reason to play with fire here. Renaming the new function `gtk_layer_set_keyboard_interactivity_type` (or something) and keeping the old one marked as deprecated is not a great burden.

Reply:
Makes sense.

Reply:
Sure, I've added new functions that are called by these.